src/services/webhook.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { Transaction } from 'ethers';
+import { lastValueFrom, catchError, retry } from 'rxjs';
+import { IncomingHttpHeaders } from 'http';
+import { JsonrpcRequestBody, WebhookResponse } from 'src/entities';
+import { Webhook } from 'src/config/transactionAllowList';
+import { URL } from 'url';
+
+@Injectable()
+export class WebhookService {
+  constructor(private readonly httpService: HttpService) {}
+
+  async post(
+    ip: string,
+    headers: IncomingHttpHeaders,
+    body: JsonrpcRequestBody,
+    tx: Transaction,
+    webhook: Webhook,
+  ): Promise<WebhookResponse> {
+    let verseHeaders: Record<string, string> = {};
+    if (headers['host']) {
+      verseHeaders['host'] = headers['host'];
+    }
+    if (headers['user-agent']) {
+      verseHeaders['user-agent'] = headers['user-agent'];
+    }
+
+    if (webhook.headers) {
+      verseHeaders = Object.assign(verseHeaders, webhook.headers);
+    }
+
+    const data = webhook.parse ? tx : body;
+
+    const webhookBody = {
+      request: {
+        clientIp: ip,
+        headers: verseHeaders,
+      },
+      data: data,
+    };
+
+    const webhookUrl = new URL(webhook.url);
+
+    const axiosConfig = {
+      timeout: webhook.timeout,
+      headers: {
+        host: webhookUrl.hostname,
+      },
+    };
+
+    const res = await lastValueFrom(
+      this.httpService.post(webhook.url, webhookBody, axiosConfig).pipe(
+        // when response status is 400 or higher
+        catchError((e) => {
+          throw e;
```

Comment:
Proxy hangs when webhook call fails. You should catch this error somewhere.

Reply:
Thanks.

I modified it to catch the webhook error and return an error response.
https://github.com/oasysgames/verse-proxy/pull/15/commits/3d47e6bac62b088f06a3507e0104217a180609da